From 1f1eb682624fbb53b54428ecd6c5bba158f63c4b Mon Sep 17 00:00:00 2001
From: Steve Dickson <SteveD@redhat.com>
Date: Thu, 26 Jun 2014 15:54:07 -0400
Subject: [fs] nfs: Handle allocation errors correctly in filelayout_alloc_layout_hdr()

Message-id: <1403798077-28799-49-git-send-email-steved@redhat.com>
Patchwork-id: 84917
O-Subject: [RHEL7.1 PATCH 48/78] pNFS: Handle allocation errors correctly in filelayout_alloc_layout_hdr()
Bugzilla: 1113248
RH-Acked-by: J. Bruce Fields <bfields@redhat.com>

From: Trond Myklebust <trond.myklebust@primarydata.com>

Return the NULL pointer when the allocation fails.

Reported-by: Fengguang Wu <fengguang.wu@intel.com>
Cc: <stable@vger.kernel.org> # 3.5.x
Signed-off-by: Trond Myklebust <trond.myklebust@primarydata.com>
Upstream Id: 6df200f5d5191bdde4d2e408215383890f956781
Bugzilla: 1113248
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/nfs/nfs4filelayout.c b/fs/nfs/nfs4filelayout.c
index 63a1637..21e9211 100644
--- a/fs/nfs/nfs4filelayout.c
+++ b/fs/nfs/nfs4filelayout.c
@@ -1348,7 +1348,7 @@ filelayout_alloc_layout_hdr(struct inode *inode, gfp_t gfp_flags)
  struct nfs4_filelayout *flo;
 
  flo = kzalloc(sizeof(*flo), gfp_flags);
- return &flo->generic_hdr;
+ return flo != NULL ? &flo->generic_hdr : NULL;
 }
 
 static void
-- 
1.7.1